ksqldb-engine/src/test/java/io/confluent/ksql/structured/SchemaKStreamTest.java

```diff
@@ -146,31 +143,39 @@ public void shouldBuildSchemaForSelect() {
   @Test
   public void shouldNotRepartitionIfSameKeyField() {
     // Given:
-    final PlanNode logicalPlan = givenInitialKStreamOf(
+    givenInitialKStreamOf(
         "SELECT col0, col2, col3 FROM test1 PARTITION BY col0 EMIT CHANGES;");
-    final UserRepartitionNode repartitionNode = (UserRepartitionNode) logicalPlan.getSources().get(0).getSources().get(0);
 
     // When:
     final SchemaKStream<?> result = initialSchemaKStream
-        .selectKey(repartitionNode.getPartitionBy(), Optional.empty(), childContextStacker);
+        .selectKey(valueFormat.getFormatInfo(), new UnqualifiedColumnReferenceExp(ColumnName.of("COL0")), Optional.empty(), childContextStacker, false);
 
     // Then:
     assertThat(result, is(initialSchemaKStream));
   }
 
   @Test
-  public void shouldNotRepartitionIfRowkey() {
```

Comment:
This test is identical to `shouldNotRepartitionIfSameKeyField` above, so the duplicate has been removed.